rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -83,14 +94,40 @@ std::vector<std::shared_ptr<rosbag2_service_info_t>> Info::read_service_info(
     throw std::runtime_error("Failed to set read order on " + uri);
   }
 
+  // Service event topic name as key
   using service_analysis =
     std::unordered_map<std::string, std::shared_ptr<service_req_resp_info>>;
-
-  std::unordered_map<std::string, std::shared_ptr<rosbag2_service_info_t>> all_service_info;
   service_analysis service_process_info;
+  std::unordered_map<std::string, std::shared_ptr<rosbag2_service_info_t>> all_service_info;
+
+  // Action name as key
+  using action_analysis =
+    std::unordered_map<std::string, std::shared_ptr<action_service_req_resp_info>>;
+  action_analysis action_process_info;
+  std::unordered_map<std::string, std::shared_ptr<rosbag2_action_info_t>> all_action_info;
 
   auto all_topics_types = storage->get_all_topics_and_types();
   for (auto & t : all_topics_types) {
+    if (is_topic_related_to_action(t.name, t.type)) {
+      std::shared_ptr<rosbag2_action_info_t> action_info;
+      std::string action_name = action_topic_name_to_action_name(t.name);
+      if (all_action_info.find(action_name) == all_action_info.end()) {
+        action_info = std::make_shared<rosbag2_action_info_t>();
+        action_info->name = action_name;
+        action_info->serialization_format = t.serialization_format;
+        all_action_info.emplace(action_name, action_info);
+        action_process_info[action_name] = std::make_shared<action_service_req_resp_info>();
+      } else {
+        action_info = all_action_info[action_name];
+      }
+
+      // cancel_goal event topic and status topic cannot get type
+      if (action_info->type.empty()) {
+        action_info->type = action_topic_type_to_action_type(t.type);
+      }
+      continue;
+    }
+
     if (is_service_event_topic(t.name, t.type)) {
       auto service_info = std::make_shared<rosbag2_cpp::rosbag2_service_info_t>();
```

Comment:
Please use else-if instead of `continue` for the sake of clarity in the logic.
```suggestion
      if (action_info->type.empty()) {
        action_info->type = action_topic_type_to_action_type(t.type);
      }
    } else if (is_service_event_topic(t.name, t.type)) {
      auto service_info = std::make_shared<rosbag2_cpp::rosbag2_service_info_t>();
```